query_execution/PolicyEnforcerBase.hpp

```diff
@@ -163,6 +174,8 @@ class PolicyEnforcerBase {
   // Key = query ID, value = QueryManagerBase* for the key query.
   std::unordered_map<std::size_t, std::unique_ptr<QueryManagerBase>> admitted_queries_;
 
+  std::unordered_set<std::size_t> removed_query_ids_;
```

Comment:
`QueryInitiateMessage` is used to initiate `QueryContext` and `CatalogCache` for query execution, while `QueryInitiateResponseMessage` is the reply message regarding the finish of such initiation.